jobs/integr8ly/pds-general.yaml

```diff
@@ -0,0 +1,300 @@
+                          throw new hudson.AbortException('YOURCITY parameter is required!')
+                      } // if
+                  } // stage
+                  
+                  stage('Execute "Before" Workarounds') {
+                      // Example of workaround
+                      //  - `oc login` credentials depend on whether the integreatly is installed or not)
+                      //  - alternative is to change config to run this directly on bastion node (not cirhos-rhel7), no `oc login` is requred then
+                      // oc login https://master.${YOURCITY}.openshiftworkshop.com -u openmgr-clt -p r3dh4t1!  --insecure-skip-tls-verify
+                      // oc delete project enmasse || true
+                      sh """
+                          echo 'No workarounds at the moment'
+                      """
+                  } // stage
+                  
+                  stage ('Uninstall #1') {
+                      if(TO_DO.contains('heavy') || TO_DO.contains('uninstall')) {
+                          build job: 'uninstallation-pipeline', parameters: [
+                              string(name: 'REPOSITORY', value: "${INSTALLATION_REPOSITORY}"),
+                              string(name: 'BRANCH', value: "${INSTALLATION_BRANCH}"),
+                              string(name: 'ANSIBLE_USER', value: 'ec2-user'),
+                              string(name: 'MASTER_URLS', value: "master1.${YOURCITY}.internal"),
+                              string(name: 'BASTION_USER', value: 'ec2-user'),
+                              string(name: 'BASTION_URL', value: "bastion.${YOURCITY}.openshiftworkshop.com"),
+                              string(name: 'BASTION_PRIVATE_KEY_ID', value: 'pds-bastion-pem')]
+                              
+                          // Waiting for 5 minutes so that resources scheduled for termination are actually terminated
+                          sleep time: 5, unit: 'MINUTES'
+                      } else {
+                          println 'Uninstall #1 skipped'
+                      }
+                  } // stage
+                  
+                  stage ('Uninstall #2') {
+                      if(TO_DO.contains('heavy') || TO_DO.count('uninstall') >= 2) {
+                          build job: 'uninstallation-pipeline', parameters: [
+                              string(name: 'REPOSITORY', value: "${INSTALLATION_REPOSITORY}"),
+                              string(name: 'BRANCH', value: "${INSTALLATION_BRANCH}"),
+                              string(name: 'ANSIBLE_USER', value: 'ec2-user'),
+                              string(name: 'MASTER_URLS', value: "master1.${YOURCITY}.internal"),
+                              string(name: 'BASTION_USER', value: 'ec2-user'),
+                              string(name: 'BASTION_URL', value: "bastion.${YOURCITY}.openshiftworkshop.com"),
+                              string(name: 'BASTION_PRIVATE_KEY_ID', value: 'pds-bastion-pem')]
```

Comment:
What about to extract this array to some reusable variable since it repeats in your proposal? This is just minor suggestion.

Reply:
Yeah, I guess there are more to optimize. It is just I spent more time on it than I expected and don't want to do it in a silo any longer. I would leave it for another PR.